fontbe/src/glyphs.rs

```diff
@@ -123,33 +127,100 @@ impl Work<Context, Error> for GlyphWork {
-
         match glyph {
             CheckedGlyph::Composite { name, components } => {
                 let composite = create_composite(context, &name, default_location, &components)?;
                 context.set_glyph(name, composite.into());
             }
-            CheckedGlyph::Contour { name, contours } => {
+            CheckedGlyph::Contour { name, paths } => {
                 // Draw the default outline of our simple glyph
-                let Some(path) = contours.get(default_location) else {
+                let Some(path) = paths.get(default_location) else {
                     return Err(Error::GlyphError(ir_glyph.name.clone(), GlyphProblem::MissingDefault));
                 };
                 let base_glyph = SimpleGlyph::from_kurbo(path).map_err(|e| Error::KurboError {
                     glyph_name: self.glyph_name.clone(),
                     kurbo_problem: e,
                     context: path.to_svg(),
                 })?;
-                context.set_glyph(name, base_glyph.into());
+                context.set_glyph(name.clone(), base_glyph.into());
+
+                let point_seqs: HashMap<_, _> = paths
+                    .into_iter()
+                    .map(|(loc, bez)| (loc, points(bez)))
+                    .collect();
+
+                let deltas = var_model
+                    .deltas(&point_seqs)
+                    .map_err(|e| Error::GlyphDeltaError(self.glyph_name.clone(), e))?;
+
+                context.set_gvar_fragment(name, GvarFragment { deltas });
             }
         }
 
         Ok(())
     }
 }
 
+fn points(path: BezPath) -> Vec<Point> {
+    let mut pen = PointPen::new();
+    write_to_pen(&path, &mut pen);
+    pen.into_inner()
+}
+
+// TODO: if this approach sticks move me to write-fonts
+#[derive(Default)]
+pub struct PointPen {
+    last_move: Option<Point>,
+    points: Vec<Point>,
+}
+
+impl PointPen {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn into_inner(self) -> Vec<Point> {
+        self.points
+    }
+}
+
+fn as_kurbo_point(x: f32, y: f32) -> Point {
+    Point {
+        x: x as f64,
+        y: y as f64,
+    }
+}
+
+impl Pen for PointPen {
+    fn move_to(&mut self, x: f32, y: f32) {
+        self.last_move = Some(as_kurbo_point(x, y));
+        self.points.push(as_kurbo_point(x, y));
+    }
+
+    fn line_to(&mut self, x: f32, y: f32) {
+        self.points.push(as_kurbo_point(x, y));
+    }
+
+    fn quad_to(&mut self, cx0: f32, cy0: f32, x: f32, y: f32) {
+        self.points.push(as_kurbo_point(cx0, cy0));
+        self.points.push(as_kurbo_point(x, y));
+    }
+
+    fn curve_to(&mut self, cx0: f32, cy0: f32, cx1: f32, cy1: f32, x: f32, y: f32) {
+        self.points.push(as_kurbo_point(cx0, cy0));
+        self.points.push(as_kurbo_point(cx1, cy1));
+        self.points.push(as_kurbo_point(x, y));
+    }
+
+    fn close(&mut self) {
+        if let Some(last_move) = self.last_move {
+            self.points.push(last_move);
```

Comment:
why are you unconditionally adding this extra move point for closed contours? this deviates from the way fonttools 
converter pens (i.e. `fontTools.pointPen.{PointToSegmentPen,SegmentToPointPen}`) translate between segments and points...
Is that because you want to make implied closing line always explicit?

Reply:
E.g. say a contour has a triangle with three on-curve points at each corner. After converting it to a BezPath with our GlyphPathBuilder and then back to points using your current PointPen, we end up with an additional point which was not there in the source!

```rust

#[cfg(test)]
mod tests {
    use fontir::ir::GlyphPathBuilder;
    use kurbo::Point;

    use super::points;

    #[test]
    fn test_roundtrip_points_to_segments_to_points() {
        let mut path_builder = GlyphPathBuilder::new("a".into());
        path_builder.line_to((0.0, 0.0)).unwrap();
        path_builder.line_to((1.0, 0.0)).unwrap();
        path_builder.line_to((1.0, 1.0)).unwrap();
        path_builder.close_path().unwrap();
        let path = path_builder.build();
        println!("{:?}", path);
        let path_points = points(path);
        println!("{:?}", path_points);

        assert_eq!(
            path_points,
            vec![
                Point::new(0.0, 0.0),
                Point::new(1.0, 0.0),
                Point::new(1.0, 1.0),
            ]
        );
    }
}
```

```sh
$ cargo test test_roundtrip_points
   Compiling fontbe v0.0.1 (/usr/local/google/home/clupo/oss/fontmake-rs/fontbe)
   Compiling fontc v0.0.1 (/usr/local/google/home/clupo/oss/fontmake-rs/fontc)
    Finished test [unoptimized + debuginfo] target(s) in 9.16s
     Running unittests src/lib.rs (target/debug/deps/fontbe-e2c6386d7cb37542)

running 1 test
test glyphs::tests::test_roundtrip_points_to_segments_to_points ... FAILED

failures:

---- glyphs::tests::test_roundtrip_points_to_segments_to_points stdout ----
BezPath([MoveTo((0.0, 0.0)), LineTo((1.0, 0.0)), LineTo((1.0, 1.0)), ClosePath])
[(0.0, 0.0), (1.0, 0.0), (1.0, 1.0), (0.0, 0.0)]
thread 'glyphs::tests::test_roundtrip_points_to_segments_to_points' panicked at 'assertion failed: `(left == right)`
  left: `[(0.0, 0.0), (1.0, 0.0), (1.0, 1.0), (0.0, 0.0)]`,
 right: `[(0.0, 0.0), (1.0, 0.0), (1.0, 1.0)]`', fontbe/src/glyphs.rs:632:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    glyphs::tests::test_roundtrip_points_to_segments_to_points

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 9 filtered out; finished in 0.00s

error: test failed, to rerun pass `-p fontbe --lib`
```

Reply:
please take a look at this wip branch of mine (based off yours) for a way to resolve the issue of roundtripping points => BezPath => points.
To match fonttools behavior, we need to change the GlyphPathBuilder to emit the implied closing line, and then in your PointPen (or in the SimpleGlyph::from_kurbo, wherever we decide to do this), we need to pop the last point of a closed contour when it's the same as the move point:

https://github.com/googlefonts/fontmake-rs/compare/gvar2...gvar3-wip?expand=1

Reply:
(I don't plan to have that branch merged, feel free to cherry pick or adapt from it)

Reply:
I actually extracted some commits from that gvar3-wip branch of mine and made https://github.com/googlefonts/fontmake-rs/pull/239

Reply:
Ha, by the time I read it you fixed it, that's awesome. #239 lgtm, I'm going to rebase on your branch.